python/cudf/cudf/core/column/column.py

```diff
@@ -472,9 +458,24 @@ def __setitem__(self, key, value):
 
         if isinstance(key, slice):
             key_start, key_stop, key_stride = key.indices(len(self))
-            if key_stride != 1:
-                raise NotImplementedError("Stride not supported in slice")
-            nelem = abs(key_stop - key_start)
+            if key_start < 0:
+                key_start = key_start + len(self)
+            if key_stop < 0:
+                key_stop = key_stop + len(self)
+            if key_start >= key_stop:
+                return self.copy()
+            if key_stride != 1 or key_stride is not None or is_scalar(value):
+                key = as_column(
+                    cudautils.arange(
```

Comment:
Moving to `cupy` in #4351:
```suggestion
                    cupy.arange(
```